libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -129,9 +134,70 @@ void HTTPClient::clear()
 }
 
 
+/**
+ * parsing the url for all needed parameters
+ * @param client Client&
+ * @param url String
+ * @param https bool
+ * @return success bool
+ */
+bool HTTPClient::begin(WiFiClient &client, String url) {
+#ifdef KEEP_PRESENT_API
+    _tcpDeprecated.reset(nullptr);
+    _transportTraits.reset(nullptr);
+#endif
+    _tcp = &client;
+
+    // check for : (http: or https:)
+    int index = url.indexOf(':');
+    if(index < 0) {
+        DEBUG_HTTPCLIENT("[HTTP-Client][begin] failed to parse protocol\n");
+        return false;
+    }
+
+    String protocol = url.substring(0, index);
+    if(protocol != "http" && protocol != "https") {
+        DEBUG_HTTPCLIENT("[HTTP-Client][begin] unknown protocol '%s'\n", protocol.c_str());
+        return false;
+    }
+
+    _port = (protocol == "https" ? 443 : 80);
```

Comment:
port may be given in the URL (`https://site.company.com:55566/`), suggest at least adding a comment here that parsing the port is not implemented yet.

Reply:
port is implemented. In line 164 _port is set to 443 or 80, depending on https or not. In the next line, beginInternal() is called. This sets _port to the value provided in the URL,if any is provided.